src/main/groovy/co/tala/performance/flo/FloRunna.groovy

```diff
@@ -56,81 +56,48 @@ class FloRunna<T> {
             logger.debug("iterations per thread: $iterationsPerThread, threads: $threads")
 
-            while (currentIteration <= iterations) {
+            while (getElapsed() < duration || currentIteration <= iterations) {
                 final long elapsed = getElapsed()
                 final int activeThreadCount = elapsed > rampup ? threads : (Math.ceil(threads * (elapsed / rampup))).toInteger()
                 // each thread gets a certain chunk of work to do
                 AtomicInteger iterationChunkCount = new AtomicInteger(1)
-                while (parallels.activeThreadCount <= activeThreadCount) {
+
+                while (parallels.activeThreadCount < activeThreadCount) {
                     parallels.runAsync {
-                        // additional check on currentIteration for times when the iterations do not break
-                        // down into nice chucks with the number of threads under test. So if we have 5
-                        // iterations, but only 3 threads, the third thread will only need 1 iteration.
-                        while(iterationChunkCount.get() <= iterationsPerThread && (currentIteration <= iterations)) {
-                            currentIteration.set(currentIteration.incrementAndGet())
-                            iterationChunkCount.set(iterationChunkCount.incrementAndGet())
+                        if (iterations > 0) {
+                            // additional check on currentIteration for times when the iterations do not break
+                            // down into nice chucks with the number of threads under test. So if we have 5
+                            // iterations, but only 3 threads, the third thread will only need 1 iteration.
+                            while (iterationChunkCount.get() <= iterationsPerThread && (currentIteration <= iterations)) {
+                                currentIteration.set(currentIteration.incrementAndGet())
+                                iterationChunkCount.set(iterationChunkCount.incrementAndGet())
+                                WorkFlo<T> workFlo = workFloClosure(new WorkFloBuilder<T>())
+                                flotility.executeWorkFlo(workFlo)
+                            }
+                        } else {
+                            // basically running only on duration and not utilizing iterations
                             WorkFlo<T> workFlo = workFloClosure(new WorkFloBuilder<T>())
                             flotility.executeWorkFlo(workFlo)
                         }
                     }
                 }
+
                 Thread.sleep(10)
-                if (currentIteration <= iterations) {
-                    logger.debug("currentIteration: $currentIteration of iterations: $iterations")
+                if (iterations > 0) {
+                    if (currentIteration <= iterations) {
+                        logger.debug("currentIteration: $currentIteration of iterations: $iterations")
```

Comment:
This may pollute the logs. Can we remove the logs? We can always use debugger if there is an issue.

Reply:
I can remove, but will only pollute if you enable debug.